ddtrace/provider.py

```diff
@@ -55,33 +60,26 @@ class DefaultContextProvider(BaseContextProvider):
 
     def __init__(self):
         # type: () -> None
-        _DD_CONTEXTVAR.set(None)
+        self._contextvar = contextvars.ContextVar(
```

Comment:
https://docs.python.org/3/library/contextvars.html#contextvars.ContextVar

> Important: Context Variables should be created at the top module level and never in closures. Context objects hold strong references to context variables which prevents context variables from being properly garbage collected.

Reply:
Hum I'd guess that'd be only true if you self-reference?

Anyhow, I had to revert this because it breaks the test in tornado and I don't know. I suspect it's a bug where there's an assumption that there's always only one context/tracer, but not sure where.

Reply:
Yeah it's probably fine in our use-case since the context provider lives for as long as the tracer which lives for as long as we care. Although maybe this could result in some hard-to-explain behaviour at some point in the future.

> I suspect it's a bug where there's an assumption that there's always only one context/tracer, but not sure where.

Yeah I've seen this too unfortunately -- it can happen when tests use multiple tracers (usually the global tracer from an integration and then a custom one 😕)